src/pip/_internal/utils/misc.py

```diff
@@ -880,15 +880,29 @@ def split_auth_from_netloc(netloc):
     return netloc, user_pass
 
 
+def redact_password_from_url(url):
+    def _redact_netloc(netloc):
```

Comment:
I would pull this function out to module level so it can be tested separately. You can use the same parametrized test cases as `test_split_auth_from_netloc()`. This will combinatorially cut down the number of test cases you need to properly test `redact_password_from_url()`, since `redact_password_from_url()` will be a simple combination of `redact_netloc()` and `transform_url()`.